ssl/quic/quic_channel.c

```diff
@@ -425,6 +435,17 @@ int ossl_quic_channel_init(QUIC_CHANNEL *ch)
     return ch_init(ch);
 }
 
+void ossl_quic_channel_bind_qrx(QUIC_CHANNEL *tserver_ch, OSSL_QRX *qrx)
+{
+    if (tserver_ch->qrx == (OSSL_QRX *)-1) {
```

Comment:
This is ugly. Could you please keep qrx at NULL for tserver and just add a flag to QUIC_CHANNEL indicating it is a tserver channel?